java/core/src/main/java/sleeper/core/statestore/FileInfoStore.java

```diff
@@ -132,6 +132,19 @@ void atomicallyUpdateJobStatusOfFiles(String jobId, List<FileInfo> fileInfos)
      */
     Map<String, List<String>> getPartitionToActiveFilesMap() throws StateStoreException;
 
+    /**
+     * Returns a count of the number of {@link FileInfo}s referencing a physical file.
+     *
+     * @param filename The filename to find the number of references to
+     * @return a count of the number of {@link FileInfo}s referencing a physical file.
+     */
+
+    long getFileReferenceCount(String filename) throws StateStoreException;
+
+    default long getFileReferenceCount(FileInfo fileInfo) throws StateStoreException {
+        return getFileReferenceCount(fileInfo.getFilename());
+    }
```

Comment:
I think in practice the cases we'll care about the file reference counts are just GC and reporting. For GC, we'll want to get the list of files that have no reference counts, and update the store to say which ones we've deleted. For reporting, we'll want to get a model of all the active references for each file. We need to define the methods on here for that.

Seeing this I'm not even sure whether the FileReferenceCount model is needed.